### ISSUE TITLE
Bug: Ensure change events are only emitted on user interaction

### ISSUE THREAD

**Issue by jcfranco** [Esri/calcite-components#2033]
### Actual Behavior

Some components will emit a change event when `value` is set programmatically (e.g., color-picker) and also after user interaction. This would be consistent with how `<input>` works.

### Expected Behavior

Change events should only emit after the user interacts with a component and not when setting the changed property programmatically.

This aligns [with the HTML spec on input/change events](https://html.spec.whatwg.org/multipage/input.html#common-input-element-events):

> When the input and change events apply (which is the case for all input controls other than buttons and those with the type attribute in the Hidden state), the events are fired to indicate that the user has interacted with the control. The input event fires whenever the user has modified the data of the control. The change event fires when the value is committed, if that makes sense for the control, or else when the control loses focus. In all cases, the input event comes before the corresponding change event (if any).

### Reproduction Steps

https://codepen.io/jcfranco/pen/yLgGmLK?editors=1000

##### Component/Event candidates

- [x] `block-section.tsx` - already emits on user interaction only (@anveshmekala)
  - [x] `93 this.calciteBlockSectionToggle.emit();`
- [x] `block.tsx` -already emits on user interaction only (@anveshmekala)
  - [ ] `150 this.calciteBlockToggle.emit();`
- [x] `card.tsx` – already emits on user interaction only @driskull 
  - [x] `138 this.calciteCardSelect.emit();`
- [x] `checkbox.tsx` – already emits on user interaction only
  - [x] `127 this.calciteCheckboxChange.emit();`
- [x] `color-picker-hex-input.tsx` (@jcfranco) – https://github.com/Esri/calcite-components/pull/4288
  - [x] `119 this.calciteColorPickerHexInputChange.emit();`
  - [x] `127 this.calciteColorPickerHexInputChange.emit();`
  - [x] `181 this.calciteColorPickerHexInputChange.emit();`
- [x] `color-picker.tsx` (@jcfranco) – https://github.com/Esri/calcite-components/pull/4288
  - [x] `266 this.calciteColorPickerInput.emit();`
  - [x] `268 this.calciteColorPickerChange.emit();`
  - [x] `279 this.calciteColorPickerInput.emit();`
  - [x] `281 this.calciteColorPickerChange.emit();`
  - [x] `553 this.calciteColorPickerChange.emit();`
- [x] `date-picker-day.tsx` – already emits on user interaction only @driskull 
  - [x] `86 !this.disabled && this.calciteDaySelect.emit();`
  - [x] `92 !this.disabled && this.calciteDaySelect.emit();`
- [x] `dropdown.tsx`
  - [x] `314 this.calciteDropdownSelect.emit();` - – already emits on user interaction only @driskull
- [x] `filter.tsx` - @driskull  https://github.com/Esri/calcite-components/pull/4287
  - [x] `197 this.calciteFilterChange.emit();`
- [x] `inline-editable.tsx`
  - [x] `292 this.calciteInlineEditableEnableEditingChange.emit();` – already emits on user interaction only @driskull 
  - [x] `346 this.calciteInlineEditableEditConfirm.emit();` – already emits on user interaction only @driskull 
- [x] `input-date-picker.tsx` 
  - [x] `226 this.calciteInputDatePickerChange.emit();` – already emits on user interaction only @driskull 
- [x] `input-time-picker.tsx` – already emits on user interaction only @driskull 
  - [x] `333 const changeEvent = this.calciteInputTimePickerChange.emit();`
- [x] `input.tsx` – already emits on user interaction only @driskull 
  - [x] `450 this.calciteInputChange.emit();`
  - [x] `482 this.calciteInputChange.emit();`
  - [x] `537 this.calciteInputChange.emit();`
  - [x] `723 this.calciteInputChange.emit();`
- [x] `notice.tsx`– already emits on user interaction only @driskull 
  - [x] `181 this.calciteNoticeClose.emit();`
- [x] ~`option-group.tsx`~ – emitting internal events
  - [x] `35 this.calciteOptionGroupChange.emit();`
- [x] ~`option.tsx`~ – emitting internal events
  - [x] `53 this.calciteOptionChange.emit();`
  - [x] `71 this.calciteOptionChange.emit();`
- [x] `panel.tsx` @driskull https://github.com/Esri/calcite-components/pull/4284
  - [x] `54 this.calcitePanelDismissedChange.emit();`
  - [x] `221 this.calcitePanelBackClick.emit();`
- [x] `pick-list-item.tsx` @driskull 
  - [x] `46 this.calciteListItemPropsChange.emit();` - internal event. 
  - [x] `77 this.calciteListItemPropsChange.emit();` - internal event. 
  - [x] `87 this.calciteListItemPropsChange.emit();` - internal event. 
  - [ ] `241 this.calciteListItemRemove.emit();` – already emits on user interaction only
- [x] `popover.tsx` @driskull @jcfranco 
  - [x] `388 this.open ? this.calcitePopoverOpen.emit() : this.calcitePopoverClose.emit();`
- [x] `radio-button.tsx` @driskull
  - [x] `163 this.calciteRadioButtonChange.emit();` – already emits on user interaction only
  - [x] `188 this.calciteRadioButtonChange.emit();` – already emits on user interaction only
  - [x] `211 this.calciteInternalRadioButtonCheckedChange.emit();` - Internal only.
- [x] `radio-group-item.tsx` - this is an internal event. @driskull 
  - [x] `43 this.calciteRadioGroupItemChange.emit();`
- [x] `select.tsx`– already emits on user interaction only @driskull 
  - [x] `328 this.calciteSelectChange.emit();`
- [x] `shell-panel.tsx` @driskull https://github.com/Esri/calcite-components/pull/4286
  - [x] `46 this.calciteShellPanelToggle.emit();`
- [x] `slider.tsx`
  - [x] `1014 this.calciteSliderInput.emit();` – already emits on user interaction only @driskull 
  - [x] `1015 this.calciteSliderUpdate.emit();` - deprecated.
  - [x] `1019 this.calciteSliderChange.emit();` – already emits on user interaction only @driskull 
- [x] `sortable-list.tsx` - already emits on user interaction only @driskull 
  - [x] `180 this.calciteListOrderChange.emit();`
- [x] `tile-select.tsx` – already emits on user interaction only @driskull 
  - [x] `144 this.calciteTileSelectChange.emit();`
  - [x] `164 this.calciteTileSelectChange.emit();`
- [x] `time-picker.tsx` – already emits on user interaction only @driskull 
  - [x] `629 this.calciteTimePickerChange.emit();`
  - [x] `670 this.calciteTimePickerChange.emit();`
- [x] `tip-manager.tsx` @driskull  https://github.com/Esri/calcite-components/pull/4285
  - [x] `40 this.calciteTipManagerToggle.emit();`
  - [x] `175 this.calciteTipManagerToggle.emit();`

##### Things to look out for

- Events emitted in a prop watcher may need to be moved elsewhere to avoid firing when props are programmatically modified
- `Change`-suffixed events are most likely emitted on both user and programmatic changes
- Change and input events should be prioritized in the first pass.
- Internal events can be refactored in a later milestone, the priority is for public ones
- Deprecated events (current or new ones) will not be modified

#### Related issues

https://github.com/Esri/calcite-components/issues/3453
https://github.com/Esri/calcite-components/issues/3191

**Comment by jcfranco** [comment]
Moving to next milestone based on yesterday's meeting. We'll have a meeting next week to discuss the pattern we want to adopt moving forward.

**Comment by jcfranco** [comment]
Depends on #2125.

**Comment by jcfranco** [comment]
Updated description with note on HTML spec on input/change events.

**Comment by paulcpederson** [comment]
#4229  I think is related to this as well.

**Comment by jcfranco** [comment]
Change/Input-related events were installed. I'll create a separate issue for the remaining non-change/input events and anything we may have missed.

**Comment by benelan** [comment]
Verified on `next` that the specified events don't emit when changed programmatically in color-picker, filter, and panel. For shell-panel, `calciteShellPanelToggle ` does still emit when changed programmatically, but the event deprecated so that's okay. I Skipped verifying `tip-manager` since the whole component was recently deprecated.
